src/triage/tracking.py

```diff
@@ -0,0 +1,282 @@
+import datetime
+import platform
+import getpass
+import os
+import requests
+import subprocess
+import logging
+from functools import wraps
+from triage.util.db import scoped_session
+from contextlib import contextmanager
+from triage import __version__
+
+
+from triage.component.results_schema import ExperimentRun, ExperimentRunStatus
+
+
+def infer_git_hash():
+    """Attempt to infer the git hash of the repository in the current working directory
+
+    Returns: Either the 'git rev-parse HEAD' output or None
+    """
+    try:
+        git_hash = subprocess.check_output(['git', 'rev-parse', 'HEAD']).strip().decode('utf-8')
```

Comment:
You _might_ prefer something like this:

```
$ git describe --always --tags
v3.1.1-9-g934179b
```

That way, if there are tags, it bakes those in. (If not, with the `--always` flag, you'll still get the hash, if need be.)